docs/source/assets/js/theme.js

```diff
@@ -0,0 +1,49 @@
+// Calcite docs theme switcher
+(function () {
+  var localStorageItem = 'calciteWebTheme';
+  var themeDarkStylesheet = document.getElementById('calcite-theme-dark');
+  var themeLightStylesheet = document.getElementById('calcite-theme-light');
+  var themeDarkAnimationClass = 'theme-dark-fade-in';
+  var themeLightAnimationClass = 'theme-light-fade-in';
+  var themeToggleEls = document.getElementsByClassName('docs-theme-toggle');
+  var themeToggles = [];
+
+  // ie array wrangling
+  for (var i = 0; i < themeToggleEls.length; i++) { themeToggles.push(themeToggleEls[i]); }
+
+  document.addEventListener('DOMContentLoaded', setThemeOnLoad);
+  themeToggles.forEach(function (el) { return el.addEventListener('click', changeTheme); });
+
+  // set disabled attribute on load here as well as in markup for ff
+  function setThemeOnLoad() {
+    if (localStorage.getItem(localStorageItem) === 'dark') {
```

Comment:
It doesn't really matter because this is doc code, but you could definitely make this less verbose

```
var isDark = localStorage.getItem(localStorageItem) === 'dark';
themeDarkStylesheet.disabled = !isDark;
themeLightStylesheet.disabled = isDark;
themeToggles.forEach(function (el) { return el.checked = isDark })
```